radixdlt/src/main/java/com/radixdlt/consensus/validators/ValidatorSet.java

```diff
@@ -54,39 +66,33 @@ public static ValidatorSet from(Collection<Validator> validators) {
 	}
 
 	/**
-	 * Validate the specified message hash against the specified signatures.
+	 * Create an initial validation state with no signatures for this validator set.
 	 *
-	 * @param message The message hash to verify signatures against
-	 * @param sigs The signatures to verify
-	 * @return A {@link ValidationResult}
+	 * @param anchor The hash to validate signatures against
+	 * @return An initial validation state with no signatures
 	 */
-	public ValidationResult validate(Hash message, Signatures sigs) {
-		final int threshold = threshold(this.validators.size());
-		final ImmutableList<Validator> signed = sigs.signedMessage(message).stream()
-			.map(this.validators::get)
-			.filter(Objects::nonNull)
-			.collect(ImmutableList.toImmutableList());
-		if (signed.isEmpty() || signed.size() < threshold) {
-			return ValidationResult.failure();
-		} else {
-			return ValidationResult.passed(signed);
-		}
+	public ValidationState newValidationState(Hash anchor) {
+		return ValidationState.forValidatorSet(anchor, this);
 	}
 
-	public ImmutableSet<Validator> getValidators() {
-		return validators.values();
+	public boolean containsKey(ECPublicKey key) {
+		return validators.containsKey(key);
 	}
 
-	@VisibleForTesting
-	static int threshold(int n) {
-		return n - acceptableFaults(n);
+	public UInt256 getPower(Set<ECPublicKey> signedKeys) {
+		return signedKeys.stream()
+			.map(validators::get)
+			.map(Validator::getPower)
```

Comment:
Should filter out null values here?

Reply:
every validator object is required to have power even if zero

Reply:
Moved to #79